File: SeLeAs/Python/LeetCode 75/maximumAverage.py
from typing import List

class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        curr_sum = max_sum = sum(nums[:k])  # Initialize current sum and max sum with the first window
        for i in range(1, len(nums) - k + 1):  # Sliding window from index 1 to len(nums) - k
            curr_sum += nums[i + k - 1] - nums[i - 1]  # Update the sum by sliding the window
            max_sum = max(max_sum, curr_sum)  # Update max_sum if a larger sum is found
        return max_sum / k  # Return the average of the maximum sum

File: LeetCode/LeetCode_75/maximumAverage.java
public class maximumAverage {
    public double findMaxAverage(int[] nums, int k) {
        // Initialize the current sum and max sum with the first window of size k
        int currSum = 0;
        for (int i = 0; i < k; i++) {
            currSum += nums[i];
        }
        
        int maxSum = currSum;  // Set maxSum initially to currSum
        
        // Sliding window logic
        for (int i = k; i < nums.length; i++) {
            currSum += nums[i] - nums[i - k];  // Update the current sum by adding the next element and removing the previous one
            maxSum = Math.max(maxSum, currSum);  // Update maxSum if a larger sum is found
        }
        
        // Return the average of the maximum sum
        return (double) maxSum / k;
    }
}
